File: Solutions/Java/Find_all_pairs_with_a_given_sum.java
class Solution {
    public pair[] allPairs( long A[], long B[], long N, long M, long X) {
        // Your code goes here 
        ArrayList<pair> ans = new ArrayList<>();
        Arrays.sort(A);
        Arrays.sort(B);
       
        int i = 0 ;
        int j = (int)M-1 ;
        while (i<A.length && j >=0) {
           if (A[i]+B[j] == X) {
                ans.add(new pair(A[i],B[j]));
                i++;
                j--;
           }
            else if (A[i]+B[j]>X) {
                j--;
            }
            else{
                i++;
            }
        }
        pair [] arr = new pair[ans.size()];
        return ans.toArray(arr);
    }
}

File: Solutions/Python/Find_all_pairs_with_a_given_sum.py
class Solution:
    def allPairs(self, A, B, N, M, X):
        # Your code goes here 
        A.sort()
        res=[]
        for i in range(N):
            if X-A[i] in B:
                res.append([A[i], X-A[i]])
        return res 
